src/jdns/jdns_sys.c

```diff
@@ -271,8 +271,8 @@ static jdns_string_t *file_nextline(FILE *f)
 	at = 0;
 	while(1)
 	{
-		unsigned char c = fgetc(f);
-		if(feof(f))
+		unsigned char c = (unsigned char)fgetc(f); // cast is intentional
+		if(feof(f)) // checking EOF here
```

Comment:
`fgetc` returns a negative value when EOF is reached or an error is encountered. This is the only instance where the cast to unsigned char is not valid. However, thanks to the check for feof afterward this is not an issue here. The comments make sure people see that this is not an error.